packages/datetime/test/common/dateUtilsTests.tsx

```diff
@@ -221,4 +221,83 @@ describe("dateUtils", () => {
     });
+
+    describe("dateToString", () => {
+        const date = new Date(2017, Months.MAY, 9, 3, 15, 22);
+
+        it("handles null input without breaking", () => {
+            assert.doesNotThrow(() => DateUtils.dateToString(null, null, null));
+        });
+
+        it("handles 'MM/DD/YYYY'", () => {
+            const format = "MM/DD/YYYY";
+
+            const dateString = DateUtils.dateToString(date, format);
+            expect(dateString).to.equal("05/09/2017");
+        });
+
+        it("handles 'YYYY-MM-DD'", () => {
+            const format = "YYYY-MM-DD";
+
+            const dateString = DateUtils.dateToString(date, format);
+            expect(dateString).to.equal("2017-05-09");
+        });
+
+        it("handles 'YYYY-MM-DD HH:mm:ss'", () => {
+            const format = "YYYY-MM-DD HH:mm:ss";
+
+            const dateString = DateUtils.dateToString(date, format);
+            expect(dateString).to.equal("2017-05-09 03:15:22");
+        });
+    });
+
+    describe("stringToDate", () => {
+        it("handles null input without breaking", () => {
+            assert.doesNotThrow(() => DateUtils.stringToDate(null));
+        });
+
+        it("handles empty input without breaking", () => {
+            assert.doesNotThrow(() => DateUtils.stringToDate(""));
+        });
+
+        it("handles 'YYYY-MM-DD'", () => {
+            const date = DateUtils.stringToDate("2015-02-15");
+            expect(toHyphenatedDateString(date)).to.equal("2015-02-15");
+        });
+
+        it("handles improperly formatted 'YYYY-M-DD'", () => {
+            const date = DateUtils.stringToDate("2015-2-15");
+            expect(toHyphenatedDateString(date)).to.equal("2015-02-15");
+        });
+
+        it("handles 'YYYY-MM-D'", () => {
+            const date = DateUtils.stringToDate("2015-10-5");
+            expect(toHyphenatedDateString(date)).to.equal("2015-10-05");
+        });
+
+        it("handles improperly formatted  'YYYY-M-D'", () => {
+            const date = DateUtils.stringToDate("2015-2-5");
+            expect(toHyphenatedDateString(date)).to.equal("2015-02-05");
+        });
+
+        it("handles 'MM/DD/YYYY'", () => {
+            const date = DateUtils.stringToDate("12/15/2015");
+            expect(toHyphenatedDateString(date)).to.equal("2015-12-15");
+        });
+
+        it("handles improperly formatted  'M/DD/YYYY'", () => {
+            const date = DateUtils.stringToDate("2/15/2015");
+            expect(toHyphenatedDateString(date)).to.equal("2015-02-15");
+        });
+
+        it("handles improperly formatted  'MM/D/YYYY'", () => {
+            const date = DateUtils.stringToDate("12/5/2015");
+            expect(toHyphenatedDateString(date)).to.equal("2015-12-05");
+        });
+
+        it("handles improperly formatted  'M/D/YYYY'", () => {
+            const date = DateUtils.stringToDate("2/5/2015");
+            expect(toHyphenatedDateString(date)).to.equal("2015-02-05");
+        });
+    });
```

Comment:
Also handles dates with no hyphens. Would prefer a check for all those weird cases so we know it's intentional.